File: _example/web.py
import os
from http.server import BaseHTTPRequestHandler, ThreadingHTTPServer


class Handler(BaseHTTPRequestHandler):
    def do_GET(self):
        body = f"hello {os.environ.get('AUTHOR', '')}".encode("utf-8")
        self.send_response(200)
        self.send_header("Content-Type", "text/plain; charset=utf-8")
        self.send_header("Content-Length", str(len(body)))
        self.end_headers()
        self.wfile.write(body)

    def log_message(self, format, *args):
        pass


port = int(os.environ.get("PORT", "8000"))
ThreadingHTTPServer(("0.0.0.0", port), Handler).serve_forever()

File: _example/web.rb
require "webrick"

port = Integer(ENV.fetch("PORT", "4567"))

server = WEBrick::HTTPServer.new(
  Port: port,
  BindAddress: "0.0.0.0",
  AccessLog: [],
  Logger: WEBrick::Log.new($stderr, WEBrick::Log::FATAL)
)

server.mount_proc "/" do |_req, res|
  res.status = 200
  res["Content-Type"] = "text/plain; charset=utf-8"
  res.body = "hello #{ENV.fetch("AUTHOR", "")}"
end

trap("INT") { server.shutdown }
trap("TERM") { server.shutdown }

server.start
